django/cantusdb_project/main_app/views/views.py

```diff
@@ -959,6 +966,36 @@ def redirect_documents(request) -> HttpResponse:
     return redirect(new_path)
 
 
+def redirect_chant_list(request) -> HttpResponse:
+    source_id: str = request.GET.get("source")
+    if source_id is None:
+        # source parameter must be provided
+        raise BadRequest("Source parameter must be provided")
+
+    base_url: str = reverse("chant-list", args=[source_id])
+
+    # optional search params
+    feast_id: str = request.GET.get("feast")
+    genre_id: str = request.GET.get("genre")
+    folio: str = request.GET.get("folio")
+    search_text: str = request.GET.get("search_text")
+
+    params: dict = {
```

Comment:
You should move the dictionary construction outside of the comprehension. It's clearer, and actually takes fewer lines.

```
d = {
  "feast": feast_id,
  "genre": genre_id,
  "folio": folio,
  "search_text": search_text,
}
params = {k: v for k, v in d.items() if v is not None}